scripts/tui-pty-smoke.py

```
@@ -0,0 +1,434 @@
+            while time.monotonic() < end:
+                drain()
+                waited, status = os.waitpid(pid, os.WNOHANG)
+                if waited == pid:
+                    drain()
+                    break
+                time.sleep(0.05)
+            if status is None:
+                os.kill(pid, signal.SIGTERM)
+                _, status = os.waitpid(pid, 0)
+            if not os.WIFEXITED(status) or os.WEXITSTATUS(status) != 0:
+                raise SystemExit(f"TUI exited unsuccessfully: status={status} output={bytes(output)!r}")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)